### ISSUE TITLE
how to set custom path for labels? please help me

### ISSUE THREAD

**Issue by J-LINC** [eriklindernoren/PyTorch-YOLOv3#819]
How can I modify the path of labels? For example, I want to divide the labels of the training set and the verification set into two subfolders, such as custom/tables ->custom/BDD100K/labels/train and custom/BDD100K/labels/val. It seems that the labels can only be placed in one folder

**Comment by jaagut** [comment]
You can do this, by changing the path in your `<config-file>.data`. See [this example file](https://github.com/eriklindernoren/PyTorch-YOLOv3/blob/master/config/custom.data).
This file points to two `.txt` files, one for training and one for validation/verification.

In the example, the configuration points to [this `train.txt`](https://github.com/eriklindernoren/PyTorch-YOLOv3/blob/master/data/custom/train.txt) and [this `valid.txt`](https://github.com/eriklindernoren/PyTorch-YOLOv3/blob/master/data/custom/valid.txt).
Currently, each of those only contains one entry. The purpose of these `train.txt` and `valid.txt` files is to point to the image files that comprise your train and verification datasets.
Each image-path in a new line.

For further information, read [here](https://github.com/eriklindernoren/PyTorch-YOLOv3#train-on-custom-dataset), especially [here](https://github.com/eriklindernoren/PyTorch-YOLOv3#define-train-and-validation-sets)

I hope, this helped. If you have further questions, please don't hesitate to ask.

**Comment by J-LINC** [comment]
First of all, thank you for your reply. I don't know if I made a mistake. EMmm, you mentioned the path for setting up the training set and verification set images, but I meant the path for the labels. In the original text, it was pointed out that the training set and verification set labels need to be placed in the labels folder, which means they are mixed together, but I want to separate the training and verification subfolders under the labels folder to store their respective label files

**Comment by Flova** [comment]
The train and validation datasets are completely independent. They are defined by the txt files that contain relative or absolute paths to the individual images (if you use relative ones make sure to start at the correct folder). You just need to make sure, that the images are in a folder called `images` and the labels are in a folder called `labels`. We look for each of the images paths in the respective txt also into the labels folder and search for a file with the same base name and the txt file extension. You should therefore be able to place all images and labels into the same folders or different ones as long as the naming follows the pattern I described.

**Comment by J-LINC** [comment]
> 

This is the answer I want. By the way, Thank you very much for your excellent code!!!!
